src/storage/ipfsStore.ts

```diff
@@ -30,84 +51,97 @@ export class IPFSSTORE<CID = any, IPFSCLIENT = any>
-                // To rebuild the tree we assume the data is found 
-                // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                for (const [secondKey, secondKeyValue] of Object.entries(
-                    value.value[jsonKey],
-                )) {
-                    // If a tree exists we denominate the start of the object with a "/"
-                    if (secondKey.includes("/")) {
-                        const newCID = value.value[jsonKey][secondKey];
-                        const branch = await this.ipfsClient.dag.get(
-                            newCID,
-                        );
-                        combinedTree = Object.assign(
-                            combinedTree,
-                            branch.value,
-                        );
-                    // If an object does not have it and is not the ".zarray" or ".zattrs" then no tree exists
-                    } else if (secondKey !== ".zarray" && secondKey !== ".zattrs") {
-                        // assign to directory for later returns
-                        this.directory = value.value[jsonKey];
-                        // Ensure a codec is loaded
-                        try {
-                            if (value.value[".zmetadata"].metadata[`${jsonKey}/.zarray`].compressor.id === "zlib") {
-                                addCodec(Zlib.codecId, () => Zlib);
-                            }
-                            if (value.value[".zmetadata"].metadata[`${jsonKey}/.zarray`].compressor.id === "blosc") {
-                                addCodec(Zlib.codecId, () => Blosc);
-                            } 
-                        // eslint-disable-next-line no-empty
-                        } catch (error) {}
-                        return value.value[".zmetadata"].metadata[`${jsonKey}/.zarray`];
+                const splitItems = item.split("/");
+                // This is used to get the .zarray object
+                // In the case of a nested array, we need to get the parent object
+                // and so we have the directory in case it is not hamt
+                let objectValue = response.value;
+                let objectValueParent = response.value;
+                for (let i = 0; i < splitItems.length; i += 1) {
+                    if (splitItems[0] === ".zarray") {
+                        objectValue = response.value[splitItems[i]];
+                        break;
                     }
+                    if (i > 0) {
+                        objectValueParent =
+                            objectValueParent[splitItems[i - 1]];
+                    }
+                    objectValue = objectValue[splitItems[i]];
+                }
+                // now check if using hamt
+                if (response.value.hamt) {
+                    this.hamt = true;
+                    // if there is a hamt, load it
+                    const hamtOptions: any = { blockHasher, blockCodec };
```

Comment:
should have a type associated as well

Reply:
i tried every trick but the ipld loader has a typescript issue it seems. These are the only two required but i keep getting typescript errors without the `any` since they say `bitWidth` and `bucketSize` are required but in documentation and code they aren't required. Works just fine without the bitwidth and bucketSize

Reply:
Looks like it's required here https://github.com/rvagg/js-ipld-hashmap/blob/22f44815bb230d56d72a71957c99b6ae03c1b6ac/interface.ts#L40 https://github.com/rvagg/js-ipld-hashmap/blob/22f44815bb230d56d72a71957c99b6ae03c1b6ac/types/interface.d.ts  no? Maybe create an issue on the repo and then maybe subsequent PR? We can also reach out to Rod he's a really great guy/extremely responsive (we had a video chat with him as well)

Reply:
Since we are loading with cid, we don't need it
![Screenshot 2023-03-29 at 8 41 56 AM](https://user-images.githubusercontent.com/37487508/228447903-30926ce6-e654-4aa8-9752-bc3585115dc1.png)

Reply:
created an issue and subsequent pull request to resolve issue

Reply:
Perfect! Linking it here for posterity 

Issue: https://github.com/rvagg/js-ipld-hashmap/issues/55
PR: https://github.com/rvagg/js-ipld-hashmap/pull/56